components/odh-notebook-controller/controllers/notebook_webhook.go

```diff
@@ -261,6 +256,12 @@ func (w *NotebookWebhook) Handle(ctx context.Context, req admission.Request) adm
 		if err != nil {
 			return admission.Errored(http.StatusInternalServerError, err)
 		}
+
+		// Mount ca bundle on notebook creation and update
+		err = CheckAndMountCACertBundle(ctx, w.Client, notebook, log)
+		if err != nil {
+			return admission.Errored(http.StatusInternalServerError, err)
+		}
```

Comment:
My first impression of this is that this is triggered only when the Notebook CR is changed somehow. But the original issue is talking also about the plain workbench restart (https://issues.redhat.com/browse/RHOAIENG-7327) - so it covers also this case?

Reply:
Looks ok to me, starting the notebook updates it, because starting/stopping (and apparently restarting!, I did not know the CR supports restarting, that is not exposed in the dashboard ui) is done through setting annotations on the CR

https://github.com/opendatahub-io/kubeflow/blob/5ad9c05f09f83cae273d2adc3bcae537d6bdc38d/components/notebook-controller/controllers/notebook_controller.go#L410

Reply:
Correct, currently, I have an ose-cli cronjob running every evening that scales down the statefulsets / stops the notebooks in all namespaces (cause users forget and we don't have long-running ones):

```
currentdatetimenotebook=$(date '+%Y-%m-%dT%H:%M:%SZ');
oc patch notebook $notebook -n "$ds_ns" --type="json" -p="[{\"op\": \"add\", \"path\": \"/metadata/annotations/kubeflow-resource-stopped\", \"value\":\"$currentdatetimenotebook\"}]";
```

Got that hint from dashboard workbench slider GUI ...

and yes, restarting is not part of dashboard logic currently on the part of notebooks, only start / stop i.e. omitting / deleting kubeflow-resource-stopped leads to start.

https://github.com/search?q=repo%3Aopendatahub-io%2Fodh-dashboard%20%22kubeflow-resource-stopped%22&type=code